test/bats/e2e-provider.bats

```diff
@@ -0,0 +1,130 @@
+
+  kubectl apply -n test-v1alpha1 -f $BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml
+  kubectl wait --for condition=established -n test-v1alpha1 --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/e2e-providerspc -n test-v1alpha1 -o yaml | grep e2e-providerspc"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+}
+
+@test "[v1alpha1] deploy e2e-providerspc secretsync crd" {
+  # Create the SPC
+  kubectl apply -n test-v1alpha1 -f $BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml
+  kubectl wait --for condition=established -n test-v1alpha1 --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/e2e-providerspc -n test-v1alpha1 -o yaml | grep e2e-providerspc"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  # Create the SecretSync
+  kubectl apply -n test-v1alpha1 -f $BATS_RESOURCE_MANIFESTS_DIR/e2e-secret-sync.yaml
+  kubectl wait --for condition=established -n test-v1alpha1 --timeout=60s crd/secretsyncs.secret-sync.x-k8s.io
+
+  cmd="kubectl get secretsyncs.secret-sync.x-k8s.io/sse2esecret -n test-v1alpha1 -o yaml | grep sse2esecret"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  # Retrieve the secret 
+  cmd="kubectl get secret sse2esecret -n test-v1alpha1 -o yaml | grep 'apiVersion: secret-sync.x-k8s.io/v1alpha1'"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  # Delete the secret
+  kubectl delete secretsync sse2esecret -n test-v1alpha1
+}
+
+@test "SecretProviderClass and SecretSync are deployed in different namespaces" {
+  # Create namespaces
+  kubectl create namespace spc-namespace --dry-run=client -o yaml | kubectl apply -f -
+  kubectl create namespace ss-namespace --dry-run=client -o yaml | kubectl apply -f -
+
+  # Deploy the SecretProviderClass in spc-namespace
+  kubectl apply -n spc-namespace -f $BATS_RESOURCE_MANIFESTS_DIR/e2e-providerspc.yaml 
+  kubectl wait --for condition=established -n spc-namespace --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/e2e-providerspc -n spc-namespace -o yaml | grep e2e-providerspc"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  # Deploy the SecretSync in ss-namespace
+  kubectl apply -n ss-namespace -f $BATS_RESOURCE_MANIFESTS_DIR/e2e-secret-sync.yaml
+  kubectl wait --for condition=established -n ss-namespace --timeout=60s crd/secretsyncs.secret-sync.x-k8s.io
+
+  cmd="kubectl get secretsyncs.secret-sync.x-k8s.io/sse2esecret -n ss-namespace -o yaml | grep sse2esecret"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  # Check that the secret is not created in ss-namespace
+  cmd="kubectl get secret sse2esecret -n ss-namespace"
+  run $cmd
+  assert_failure
+}
+
+@test "Delete SecretSync and check that the secret is deleted" {
```

Comment:
Added a comment above. I don't think we need a separate test for this. I would recommend validating every required step in `@test "[v1alpha1] deploy e2e-providerspc secretsync crd"` as a sequence.